src/storage/src/sink/kafka.rs

```diff
@@ -783,50 +724,38 @@ impl KafkaSinkState {
             Ok(latest_ts)
         }
 
-        if let KafkaSinkStateEnum::Init(ProgressInitState {
```

Comment:
this `if let` hasn't really been necessary since we got rid of the optionality of `ProgressInitState` when we committed to always tracking progress